app/helpers/katello/subscription_mailer_helper.rb

```diff
@@ -0,0 +1,25 @@
+module Katello
+  module SubscriptionMailerHelper
+    include Rails.application.routes.url_helpers
+
+    def report_url
+      report_id = ReportTemplate.find_by(name: "Subscription - Entitlement Report").id
+      days_from_now = TemplateInput.find_by_name("Days from Now").id.to_s
+      mail_query = UserMailNotification.first.mail_query.to_s
+
+      params = { format: 'csv', template_id: report_id , input_values: { days_from_now => { value: mail_query} } }
+
+      provider_job_id = start_report_task(params)
+      base_url = report_data_report_template_path(report_id, job_id: provider_job_id)
+      "#{Setting[:foreman_url]}#{base_url}"
+    end
+
+    def start_report_task(params)
```

Comment:
I think this method should just return the result of `composer.schedule_rendering` that way we can gather more data if needed beyond the provider_job_id without having to change the method later on